datafusion/core/src/datasource/file_format/write/mod.rs

```diff
@@ -69,79 +66,6 @@ impl Write for SharedBuffer {
     }
 }
 
-/// Stores data needed during abortion of MultiPart writers
```

Comment:
Is it correct to say that the implications of removing `AbortableWrite`  is that if certain (larger) writes to object store fail / abort for some reason, "garbage" (unreferenced partial uploads) may be left around indefinitely on the provider? 

While I understand that some object stores (maybe all) can be configured to automatically clean up such parts, I think reverting the "try to cleanup on failure" behavior is worth reconsidering. 

I think I could be convinced with an argument like "the software can't ensure clean up anyways (for example, if it is `SIGKILL`ed) for some reason, and thus we don't even try to clean up in paths we could", but if we go that route I think we should explicitly document the behavior and rationale in comments somewhere


I think @metesynnada or @mustafasrepo  originally added this code (though I may be wrong) so perhaps they have some perspective to share

Reply:
Yeah, I think according to https://github.com/apache/arrow-datafusion/pull/9648#issuecomment-2002483969, 'garbage' cleanup will be only on cloud provider if removing `AbortableWrite` 🤔, also @devinjdangelo , is it right?

Reply:
I have mixed feelings about removing any attempt to clean up on failure.

> I think I could be convinced with an argument like "the software can't ensure clean up anyways (for example, if it is SIGKILLed) for some reason, and thus we don't even try to clean up in paths we could", but if we go that route I think we should explicitly document the behavior and rationale in comments somewhere

This argument is valid. A hardware/network fault will prevent any cleanup code we write from working, so to truly protect against partial writes and would require logic outside of DataFusion's process (e.g. on the cloud service provider side).

On the other hand, this change may be annoying for simple failures when writing to a local file system. Encountering any execution error will leave dangling files when before they often could be cleaned up.

Reply:
Local file system automatically cleans up, or at least makes a best effort to do so

Reply:
Ok, so I vote we leave the code as is (no attempt to explicitly abort in write) and add a note in the documentation. If it turns out this is an important behavior, we can add it back in

@yyy1000  can you handle adding the note in the documentation

Reply:
Sure, I added it around `create_writer` function, would it be enough?

Reply:
Looks reasonable to me --  thank you